autoencoder.py

```diff
@@ -22,51 +37,132 @@ def __init__(self, intermediate_dim, code_dim, original_dim, num_hidden_layers,
         self.num_hidden_layers = num_hidden_layers
         self.encoder_hidden_layers = []
         self.decoder_hidden_layers = []
-        """Encoder"""
+        self.learning_rate = learning_rate
+        if training is False:
+            self.learning_rate = 1e-3
+        self.optimizer = \
+            tf.optimizers.Adam(learning_rate=self.learning_rate)
+
         for layer in range(self.num_hidden_layers):
-          self.encoder_hidden_layers.append(Dense(units=self.intermediate_dim, activation=tf.nn.relu))
-          self.intermediate_dim = self.intermediate_dim - self.encoder_decrements
-        self.encoder_output_layer = Dense(units=code_dim, activation=tf.nn.relu)
-        """Decoder"""
+            self.encoder_hidden_layers.append(Dense(units=self.intermediate_dim,
+                    activation=tf.nn.relu))
+            self.intermediate_dim = self.intermediate_dim \
+                - self.encoder_decrements
+        self.encoder_output_layer = Dense(units=code_dim,
+                activation=tf.nn.relu)
         for layer in range(self.num_hidden_layers):
-          self.decoder_hidden_layers.append(Dense(units=self.code_dim, activation=tf.nn.relu))
-          self.code_dim = self.code_dim - self.decoder_increments
-        self.decoder_output_layer = Dense(units=original_dim, activation=tf.nn.relu)
-        
+            self.decoder_hidden_layers.append(Dense(units=self.code_dim,
+                    activation=tf.nn.relu))
+            self.code_dim = self.code_dim - self.decoder_increments
+        self.decoder_output_layer = Dense(units=original_dim,
+                activation=tf.nn.relu)
+
     def call(self, input_features):
-      """Encoder"""
-      activation = self.encoder_hidden_layers[0](input_features)
-      if self.num_hidden_layers > 1:
-        for layer in range(self.num_hidden_layers - 1):
-          layer = layer + 1
-          activation = self.encoder_hidden_layers[layer](activation)
-      activation = self.encoder_output_layer(activation)           
-      """Decoder"""
-      activation = self.decoder_hidden_layers[0](activation)
-      if self.num_hidden_layers > 1:
-        for layer in range(self.num_hidden_layers - 1):
-          layer = layer + 1
-          activation = self.decoder_hidden_layers[layer](activation)
-      return self.decoder_output_layer(activation)
+        """Encoder"""
+
+        activation = self.encoder_hidden_layers[0](input_features)
+        if self.num_hidden_layers > 1:
+            for layer in range(self.num_hidden_layers - 1):
```

Comment:
We can simply use it as it is, `range(self.num_hidden_layers)`.
We start with index 0 anyway (supposedly).

Reply:
Noted on this. Thank you.